### ISSUE TITLE
Siad has exited unexpectedly. Severe error: Unable to write state change to WAL: invalid argument panic: unable to append a change to the WAL, crashing to prevent corruption.  Not Synchronizing 97.3% done

### ISSUE THREAD

**Issue by summitmann** [NebulousLabs/Sia#2591]
Bug Report
==========
**Stack Trace or error message**
Sia Daemon v1.3.1
Loading...
(0/6) Loading siad...
(1/6) Loading gateway...
listen tcp :9981: bind: address already in use
goroutine 4465 [running]:
runtime/debug.Stack(0x1045e47, 0xc4215f6cc0, 0xc4215f6c88)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/NebulousLabs/Sia/build.Severe(0xc4215f6e58, 0x2, 0x2)
	/home/luke/go/src/github.com/NebulousLabs/Sia/build/critical.go:32 +0x85
github.com/NebulousLabs/Sia/persist.(*Logger).Severe(0xc4201de460, 0xc4215f6e58, 0x2, 0x2)
	/home/luke/go/src/github.com/NebulousLabs/Sia/persist/log.go:71 +0xcc
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit.func2(0xc42140b110, 0xc4200a6ae8, 0x0, 0x0, 0x0, 0x1b20b60, 0x0, 0x0)
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:196 +0x1d1
created by github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:188 +0x1e5
Severe error: ERROR: unable to create write-ahead-log: open /Users/mdhmann/Library/Application Support/Sia-UI/sia/host/contractmanager/contractmanager.wal_temp: too many open files in system
goroutine 4465 [running]:
runtime/debug.Stack(0x1045e47, 0xc4215f6cc0, 0xc4215f6c88)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/NebulousLabs/Sia/build.Severe(0xc4215f6e98, 0x2, 0x2)
	/home/luke/go/src/github.com/NebulousLabs/Sia/build/critical.go:32 +0x85
github.com/NebulousLabs/Sia/persist.(*Logger).Severe(0xc4201de460, 0xc4215f6e98, 0x2, 0x2)
	/home/luke/go/src/github.com/NebulousLabs/Sia/persist/log.go:71 +0xcc
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit.func2(0xc42140b110, 0xc4200a6ae8, 0x0, 0x0, 0x0, 0x1b20b60, 0x0, 0x0)
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:201 +0x29e
created by github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:188 +0x1e5
Severe error: Unable to properly initialize WAL file, crashing to prevent corruption: unable to write WAL metadata: invalid argument
goroutine 4465 [running]:
runtime/debug.Stack(0x1045e47, 0xc4215f6b28, 0xc4215f6af0)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/NebulousLabs/Sia/build.Severe(0xc4215f6c40, 0x2, 0x2)
	/home/luke/go/src/github.com/NebulousLabs/Sia/build/critical.go:32 +0x85
github.com/NebulousLabs/Sia/persist.(*Logger).Severe(0xc4201de460, 0xc4215f6c40, 0x2, 0x2)
	/home/luke/go/src/github.com/NebulousLabs/Sia/persist/log.go:71 +0xcc
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).appendChange(0xc4200a6ae8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlog.go:151 +0x2e9
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit.func2(0xc42140b110, 0xc4200a6ae8, 0x0, 0x0, 0x0, 0x1b20b60, 0x0, 0x0)
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:205 +0x36a
created by github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:188 +0x1e5
Severe error: Unable to write state change to WAL: invalid argument
panic: unable to append a change to the WAL, crashing to prevent corruption

goroutine 4465 [running]:
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).appendChange(0xc4200a6ae8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlog.go:152 +0x305
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit.func2(0xc42140b110, 0xc4200a6ae8, 0x0, 0x0, 0x0, 0x1b20b60, 0x0, 0x0)
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:205 +0x36a
created by github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit
	/home/luke/go/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:188 +0x1e5

**Expected Behavior**
complete syncronization.  

**Environment**
* Sia version: 1.3.1
* OS: Mac OS 10.11.6

**Any useful logs**
see above 
97.3% done

Feature Request
===============
Check to see if the feature is on the roadmap.  If it is not on the roadmap then clearly describe your feature request and why it would benefit Sia.

[Sia roadmap](https://trello.com/b/Io1dDyuI/sia-public-roadmap)

**Comment by DavidVorick** [comment]
Hi @summitmann thanks for your bug report. We've seen a couple users post this log, can you answer a few questions for me to help me figure out what's going wrong?

Was Sia running while this was happening? Part of your log doesn't make much sense to me, it's this part:

Sia Daemon v1.3.1
Loading...
(0/6) Loading siad...
(1/6) Loading gateway...
listen tcp :9981: bind: address already in use

If you are seeing that message, it ~should shut down instead of continue loading. But it seems like it continued loading. Can you confirm that Sia was running and stable for a while? Or did it stop working and display this relatively quickly?

Have you gotten Sia working in the past? Did you have any contracts? Were you using the UI, or just the daemon?

**Comment by summitmann** [comment]
Hi David!

I tried to install sia overnight and was asleep when the most of the log was produced.  
I went to be at 1 a.m. CDT with about 54% done.  Then, at 6 a.m when I woke up it showed 97% done not synchronizing message in the top right corner.  
At that point I clicked the start sia button, and that’s when that very last part of the log “address already in use part popped up.”  

Sorry I can’t answer the when did it go wrong part of your question any better.

Regarding usage.  I will use the daemon for sure.   I am also planning to use the UI, have contracts, and all of that in the near future.  

I have put my siacoins in  easysia.com <http://easysia.com/> for now.  I am not too sure it is safe.  But, until I get your wallet running properly,  i consider it better than storing coins on the bittrex exchange.

Thanks for your time, and best of luck with everything..


Mark Mann, Ph.D., Ed.D. 
mmann@summit.edu
Cell: 918-231-1147
Twitter:  @summitmann
